From f5f9565dee23a3e11a52bcc102ba03ba1ec3f9fe Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 25 Jan 2023 10:11:18 +0100
Subject: [PATCH] dt-bindings: interrupt-controller: Document the rpmsg
 interrupt controller

Document the RPMsg interrupt controller bindings.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I0467a01f393bbe5b7141f2e4501e725beccc8d8d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/286888
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 .../interrupt-controller/rpmsg,intc.yaml      | 39 +++++++++++++++++++
 1 file changed, 39 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/interrupt-controller/rpmsg,intc.yaml

diff --git a/Documentation/devicetree/bindings/interrupt-controller/rpmsg,intc.yaml b/Documentation/devicetree/bindings/interrupt-controller/rpmsg,intc.yaml
new file mode 100644
index 000000000000..15bda0b01f86
--- /dev/null
+++ b/Documentation/devicetree/bindings/interrupt-controller/rpmsg,intc.yaml
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+# Copyright (c) 2023 STMicroelectronics.
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/interrupt-controller/rpmsg,intc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: RPMSG Interrupt controller
+
+maintainers:
+  - Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
+
+allOf:
+  - $ref: /schemas/interrupt-controller.yaml#
+
+description:
+  RPMsg interrupt controller for interprocessor communication based on interrupt signaling
+
+properties:
+  compatible:
+    const: rpmsg,intc
+
+  '#interrupt-cells':
+    const: 1
+
+  interrupt-controller: true
+
+required:
+  - compatible
+  - '#interrupt-cells'
+  - interrupt-controller
+
+examples:
+  - |
+    intc_rpmsg: interrupt-controller {
+        compatible = "rpmsg,intc";
+        #interrupt-cells = <1>;
+        interrupt-controller;
+    };
-- 
2.39.5

